data_generators/xcspec_extractor.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+
+import plistlib
+import pprint
+import subprocess
+import os
+import sys
+
+
+def xcspec_from_file(path):
+    xml = subprocess.check_output(
+        ['plutil', '-convert', 'xml1', '-o', '-', '--', path])
+    plist = plistlib.loads(xml, fmt=plistlib.FMT_XML)
+    if not isinstance(plist, list):
+        plist = [plist]
+    return plist
+
+
+developer_dir = sys.argv[1]
+xcode_version = sys.argv[2]
+xcode = os.path.dirname(os.path.dirname(developer_dir))
+xcode3plusgins = os.path.join(
+    xcode, 'Contents/PlugIns/Xcode3Core.ideplugin/Contents/SharedSupport/Developer/Library/Xcode/Plug-ins')
+
+xcspecs = [
+    os.path.join(xcode3plusgins,
+                 "CoreBuildTasks.xcplugin/Contents/Resources/Ld.xcspec"),
+    os.path.join(
+        xcode3plusgins, "Clang LLVM 1.0.xcplugin/Contents/Resources/Clang LLVM 1.0.xcspec"),
+    os.path.join(xcode3plusgins,
+                 "Core Data.xcplugin/Contents/Resources/Core Data.xcspec"),
+    os.path.join(
+        xcode3plusgins, "IBCompilerPlugin.xcplugin/Contents/Resources/IBCompiler.xcspec"),
+    os.path.join(xcode3plusgins,
+                 "XCLanguageSupport.xcplugin/Contents/Resources/Swift.xcspec"),
+]
+
+a = []
+for path in xcspecs:
+    a.extend(xcspec_from_file(path))
+
+ids = (
+    "com.apple.compilers.llvm.clang.1_0",
+    "com.apple.compilers.model.coredata",
+    "com.apple.compilers.model.coredatamapping",
+    "com.apple.pbx.linkers.ld",
+    "com.apple.xcode.tools.ibtool.compiler",
+    "com.apple.xcode.tools.swift.compiler",
+)
+
+print("# Extracted from Xcode %s" % xcode_version)
```

Comment:
IIUC - `data_generators` is a manual task to run to update the xcspec, correct? Worth adding a comment to the top of the file produced with how to go about regenerating it?

Reply:
Bump